### DISCUSSION TITLE
run LAMMPS problem

### DISCUSSION THREAD

**Discussion by KK33999** [deepmodeling/deepmd-kit#4691] [Q&A] [answered]
Hello everyone, I have a question to discuss. I trained the potential function using the DeePMD-kit v3.0.2 version DPA2 descriptor, and the energy and force accuracy were also in good agreement during the dp test process. However, when I used this potential function to run LAMMPS molecular dynamics, I found that it would scatter the structure after 20 ps. The specific details of my training are: 1. Select the temperature range of 50K-500K and use VASP to run AIMD. Check that the structure of the last step is not scattered and select 80% of all data for training. 2. Use the LAMMPS template as follows. 

````
variable			t equal 50

log				log_$t.lammps  
units           		metal                                                  
boundary        		p p p           
atom_style     		atomic # full  # charge  dipole     
read_data	  		stru.data  
mass			1  A      
mass 			2  B

group			A  type 1
group			B  type 2     

pair_style	deepmd 	./graph.pth        
pair_coeff  		* *	                                 
velocity        		all create $t 23456789 dist gaussian

timestep        		0.001
fix             		1 all nvt temp $t $t 0.1 
                                               
thermo_style    	custom step temp pe ke etotal press density lx ly lz vol         
thermo          		100                                                                                         

dump			1 all custom 10 $t.dump id type x y z 

thermo			10                       

run             		500000
````

![image](https://github.com/user-attachments/assets/2d2e1e6f-cbfa-4f29-a673-f88eb0fac485)

My guess is whether AIMD data needs to be cleaned? I don't know, I hope someone can give some suggestions or similar experiences.

**Comment by njzjz** [comment, marked as answer]
Do you know the active learning? It may be just because the data is not enough.

**Reply by KK33999** [reply]
Finally, I expanded the dataset with dpgen and achieved good results. Thank you for your answer. Best wishes.

**Comment by pillowill** [comment]
May I ask how you successfully ran LAMMPS? I am using version 3.1.1 of DeepMD-kit. The tutorial I am following is at https://deepmodeling.com/articles/detail/972da94c-bc35-43b2-af20-6ca943727015/DeePMD-PyTorch. The contents of in.lammps are as follows：
# bulk water

units           metal
boundary        p p p
atom_style      atomic

neighbor        2.0 bin
neigh_modify    every 10 delay 0 check no

read_data	water.lmp
mass 		1 16
mass		2 2

# See https://deepmd.rtfd.io/lammps/ for usage
pair_style	deepmd frozen_model.pth
# If atom names (O H in this example) are not set in the pair_coeff command, the type_map defined by the training parameter will be used by default.
pair_coeff  * *	O H

velocity        all create 330.0 23456789

fix             1 all nvt temp 330.0 330.0 0.5
timestep        0.0005
thermo_style    custom step pe ke etotal temp press vol
thermo          100
dump		1 all custom 100 water.dump id type x y z

run             1000
The runtime error is displayed as follows:
(deepmd-pt) HeYu@uubntu:~/deepmd/deepmd-kit/examples/water/lmp$ lmp -in in.lammps 
LAMMPS (2 Aug 2023 - Development - 27347ce4a1)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Reading data file ...
  triclinic box = (0 0 0) to (12.4447 12.4447 12.4447) with tilt (0 0 0)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  192 atoms
  read_data CPU = 0.002 seconds
Exception: 
Unknown builtin op: deepmd::border_op.
Could not find any similar ops to deepmd::border_op. This op may not exist or may not be currently supported in TorchScript.
:
  File "code/__torch__/deepmd/pt/model/descriptor/repformers.py", line 238
      _34 = torch.tensor(real_nloc, dtype=3, device=torch.device("cpu"))
      _35 = torch.tensor(torch.sub(real_nall, real_nloc), dtype=3, device=torch.device("cpu"))
      ret = ops.deepmd.border_op(_27, _28, _29, _31, _32, g10, _33, _34, _35)
            ~~~~~~~~~~~~~~~~~~~~ <--- HERE
      g1_ext0 = torch.unsqueeze(ret[0], 0)
      if has_spin:

--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.

**Reply by iProzd** [reply]
Hi @pillowill , it appears that DeePMD-kit or LAMMPS may not be installed correctly. I recommend using the offline packages (refer to the [installation guide](https://docs.deepmodeling.com/projects/deepmd/en/stable/install/easy-install.html#install-off-line-packages) and [v3.1.1 release](https://github.com/deepmodeling/deepmd-kit/releases/tag/v3.1.1)) to ensure both are set up properly. Once installed, please ensure you freeze your checkpoint (\*.pt) into a model file (\*.pth) using this specific version before running the LAMMPS script. You may also find the [DPA2 tutorial](https://www.bohrium.com/notebooks/83525789178) and our newest [DPA3 model](https://www.aissquare.com/models/detail?pageType=models&name=DPA-3.1-3M&id=343) helpful.
